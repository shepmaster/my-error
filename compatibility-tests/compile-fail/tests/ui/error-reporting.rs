mod wrong_error_types {
    use snafu::Snafu;

    #[derive(Snafu)]
    union AUnion {
        _a: i32,
    }

    #[derive(Snafu)]
    enum TupleEnumVariant {
        Alpha(i32),
    }
}

mod other_attributes {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum Error {
        #[serde]
        UnknownVariantAttributeIsIgnored,
    }
}

mod display {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum FailedAttributeParsing {
        #[snafu(display)]
        DisplayWithoutArgument,

        #[snafu(display = 42)]
        StringStyleNotStringLiteral,
    }

    #[derive(Debug, Snafu)]
    enum InvalidGeneratedCode {
        #[snafu(display(foo()))]
        CleanStyleBadFormatString,

        #[snafu(display = "42")]
        StringStyleBadFormatString,
    }
}

mod opaque {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    struct UnitStruct;

    #[derive(Debug, Snafu)]
    struct NamedFieldStruct {
        some_field: i32,
    }

    #[derive(Debug, Snafu)]
    struct ShortTupleStruct();

    #[derive(Debug, Snafu)]
    struct LongTupleStruct(i32, i32);
}

fn main() {}
