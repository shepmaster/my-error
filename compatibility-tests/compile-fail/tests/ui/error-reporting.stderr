error: Can only derive `Snafu` for an enum or a newtype
 --> $DIR/error-reporting.rs:5:5
  |
5 |     union AUnion {
  |     ^^^^^

error: Only struct-like and unit enum variants are supported
  --> $DIR/error-reporting.rs:11:14
   |
11 |         Alpha(i32),
   |              ^^^^^

error: `snafu(display)` requires an argument
  --> $DIR/error-reporting.rs:30:17
   |
30 |         #[snafu(display)]
   |                 ^^^^^^^

error: expected string literal
  --> $DIR/error-reporting.rs:33:27
   |
33 |         #[snafu(display = 42)]
   |                           ^^

error: format argument must be a string literal
  --> $DIR/error-reporting.rs:39:25
   |
39 |         #[snafu(display(foo()))]
   |                         ^^^^^
   |
help: you might be missing a string literal to format with
   |
39 |         #[snafu(display("{}", foo()))]
   |                         ^^^^^

error: format argument must be a string literal
  --> $DIR/error-reporting.rs:42:27
   |
42 |         #[snafu(display = "42")]
   |                           ^^^^
   |
help: you might be missing a string literal to format with
   |
42 |         #[snafu(display = "{}", "42")]
   |                           ^^^^^

error: Can only derive `Snafu` for tuple structs with exactly one field
  --> $DIR/error-reporting.rs:59:5
   |
59 |     struct ShortTupleStruct();
   |     ^^^^^^

error: Can only derive `Snafu` for tuple structs with exactly one field
  --> $DIR/error-reporting.rs:62:5
   |
62 |     struct LongTupleStruct(i32, i32);
   |     ^^^^^^

error: cannot find attribute `serde` in this scope
  --> $DIR/error-reporting.rs:20:11
   |
20 |         #[serde]
   |           ^^^^^
